source/Octopus.Tentacle/Security/Certificates/CertificateEncoder.cs

```diff
@@ -186,7 +186,7 @@ static bool HasPrivateKey(X509Certificate2 certificate2)
         {
             try
             {
-                return certificate2.HasPrivateKey && certificate2.PrivateKey != null;
+                return certificate2.GetRSAPrivateKey() != null;
```

Comment:
This is the only code change I had to make. The `EphemeralKeySet` flag is inside a `#if NET472_OR_GREATER || NETCOREAPP || NETSTANDARD` condition, so it was always ignored on `net452` and `PersistKeySet` would be used. From `net472` we have `EphemeralKeySet` but it was erroring due to private key not being found. As per the MS docs on `EphemeralKeySet` support:

> Keys loaded in this manner are almost always loaded via Windows CNG. Therefore, callers must access the private key by calling extension methods, such as [cert.GetRSAPrivateKey()](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.rsacertificateextensions.getrsaprivatekey). The [X509Certificate2.PrivateKey](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2.privatekey#system-security-cryptography-x509certificates-x509certificate2-privatekey) property does not function.

I'm assuming `GetECDsaPrivateKey` doesn't come into play here. There's only reference to RSA.

Reply:
Also TODO: Add a comment that this _may_ be reverted to `.PrivateKey` once .NET framework is dropped entirely. Will use this change to trigger the final build before merge.

Reply:
Does this TODO need action?

Reply:
Yeah that was confusing. Looking at it again, no action needed.